File: python/stack.py
class Stack():

	def __init__(self):
		self.items = []
		self.len = 0
	def push(self, data):
		self.items.append(data)
		self.len += 1
	def pop(self):
		self.items.pop()
		self.len -= 1
	def isEmpty(self):
		if self.items == []:
			return True
		else:
			return False

s1 = Stack()
for i in range(20):
	s1.push(i)
print(s1.items)
s1.pop()
print(s1.items)
print(s1.isEmpty())
print(s1.len)

File: go/stack.go
package main

import "fmt"

func main() {

	var stack Stack
	stack.Push(10)
	stack.Push(20)
	stack.Push(40)
	fmt.Println(stack)
	stack.Pop()
	fmt.Println(stack)
	fmt.Println(stack.Len())

}

type Stack []interface{}

func (s *Stack) Push(val interface{}) interface{} {
	*s = append(*s, val)
	return val
}

func (s *Stack) Pop() interface{} {
	*s = (*s)[:len(*s)-1]
	return (*s)[len(*s)-1]
}
func (s *Stack) Len() int {
	return len(*s)
}
func (s *Stack) isEmpty() bool {
	if len(*s) == 0 {
		return true
	}
	return false
}
